tests/common/defender/aws_iot_tests_defender_api.c

```diff
@@ -471,16 +491,30 @@ TEST( Full_DEFENDER, SetPeriod_after_started )
 static void _verifyCallbackFunction( void * param1,
                                      AwsIotDefenderCallbackInfo_t * const pCallbackInfo )
 {
-    TEST_ASSERT_NOT_NULL( pCallbackInfo );
+    /* Copy and malloc data from pCallbackInfo to _callbackInfo. */
+    if( pCallbackInfo != NULL )
+    {
+        _callbackInfo.eventType = pCallbackInfo->eventType;
+        _callbackInfo.metricsReportLength = pCallbackInfo->metricsReportLength;
+        _callbackInfo.payloadLength = pCallbackInfo->payloadLength;
 
-    _reportRejected = pCallbackInfo->eventType == AWS_IOT_DEFENDER_METRICS_REJECTED;
+        if( _callbackInfo.payloadLength > 0 )
+        {
+            _callbackInfo.pPayload = AwsIotTest_Malloc( _callbackInfo.payloadLength );
```

Comment:
line 503, 512, 533, and 534. Symmetry is beauty. Better to put allocate/free both in test setup/teardown. Or both in local test.

Reply:
This makes sense. I can move the allocate to setup